src/net.cpp

```diff
@@ -1080,7 +1080,12 @@ void ThreadSocketHandler()
                 }
             }
         }
-        if(vNodes.size() != nPrevNodeCount) {
+        size_t vNodesSize;
+        {
+            LOCK(cs_vNodes);
+            vNodesSize = vNodes.size();
+        }
+        if(vNodesSize != nPrevNodeCount) {
             nPrevNodeCount = vNodes.size();
```

Comment:
Here is my opinion, I guess we don't need `if(vNodesSize != nPrevNodeCount) {` because at the end , `nPrevNodeCount` will be forced set to `vNode.size()`. So, I would replace old line1083-1084 to be
```
 +        {
 +            //Reset vPrevNodeCount to current...
 +            LOCK(cs_vNodes);
 +            vPrevNodeCount = vNodes.size();
 +        }
```